### PR TITLE
Migrate x-google-marketplace v1 to v2

### PR DESCRIPTION
**Pull request by tkohn**
* refactor code structure to match GCP examples
* change the apptester app, introduce tester image
* update helm chart to use the latest version
* disable serviceAccount and rbac in values.yaml
* create verify.sh script to build/push and verify changes
* update schema.yaml
  * change default endpoint
  * add tls configuration (only secretName)
  * change memory limits

### CODE REVIEW

File: apptest/tester/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM gcr.io/cloud-marketplace-tools/testrunner:0.1.4
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    curl wget \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN mkdir -p /opt/kubectl/1.14 \
```
Comment (mdonkers) [inline apptest/tester/Dockerfile]: this is pretty old Kubernetes, are there no later versions available? (I think its currently at 1.22 or 1.23 or so)

File: chart/instana-agent/Chart.yaml [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
+- email: jeroen.soeters@instana.com
+  name: JeroenSoeters
+- email: fabian.staeber@instana.com
+  name: fstab
+- email: miel.donkers@instana.com
+  name: mdonkers
+- email: dahlia.bock@instana.com
+  name: dlbock
+- email: nathan.fisher@instana.com
+  name: nfisher
```
Comment (mdonkers) [inline chart/instana-agent/Chart.yaml]: Let's remove Jon, William, Jeroen, Fabian, Dahlia and Nathan.
Only team Agent is responsible for this stuff. And Jon and Fabian have left / will leave Instana.
And add yourself to this list :-P

File: chart/instana-agent/values.yaml [hunk 3]
```diff
@@ -81,3 +170,73 @@ agent:
...
+opentelemetry:
+  enabled: false # If true, it will also apply `service.create=true`
+
+prometheus:
+  remoteWrite:
+    enabled: false # If true, it will also apply `service.create=true`
+
+serviceAccount:
+  # Specifies whether a ServiceAccount should be created
+  create: false
```
Comment (mdonkers) [inline chart/instana-agent/values.yaml:206]: This one (and the `rbac`) should not be used by the customer, right? If they enable then the deployment fails?
Perhaps then we should not expose them at all. Or put them in some separate place and make it _really_ clear customers should not change those params.

Comment (mdonkers) [reply chart/instana-agent/values.yaml:206]: Or is this handled correctly by this `schema.yaml` file and customers wouldn't touch this `values.yaml` anyhow?
(then still at least some comment might be helpful for the next time we need to update with our latest Helm chart)

**Review by mdonkers — COMMENTED** [review]
small few questions, but otherwise great work! I know it has not been easy... :-/

File: chart/instana-agent/values.yaml (hunk 3 above)
Comment (tkohn) [reply chart/instana-agent/values.yaml:206]: Should be handled by Schema because you can only change attributes which are available in the Schema.

**Review by mdonkers — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -17,34 +17,27 @@ gcloud components install docker-credential-gcr
 docker-credential-gcr configure-docker
 ```
 
-# Rules
+## Rules
 
-We are creating an app with building blocks from our existing helm chart. There are rules different rules for Helm chart and Google market app. While helm recommends RBAC and SA, that is forbidden with Google Marketplace App. ClusterRole, ClusterRoleBinding and ServiceAccount are defined in `schema.yaml`.
+We are creating an app with building blocks from our existing helm chart. 
+There are rules different rules for Helm chart and Google market app.
```
Comment (FelixMarxIBM) [inline README.md]:
```suggestion
There are different rules for Helm chart and Google market app. 
```

File: chart/instana-agent/Chart.yaml [hunk 5]
```diff
@@ -1,24 +1,22 @@
...
-    email: fabian.staeber@instana.com
-  - name: mdonkers
-    email: miel.donkers@instana.com
-  - name: dlbock
-    email: dahlia.bock@instana.com
-  - name: nfisher
-    email: nathan.fisher@instana.com
+- email: henning.treu@ibm.com
+  name: htreu
+- email: m.donkers@ibm.com
```
Comment (FelixMarxIBM) [inline chart/instana-agent/Chart.yaml]: Might make sense to remove Miel here already (sorry about that) and include myself.

**Pull request merged**